chia/data_layer/download_data.py

```diff
@@ -254,6 +263,7 @@ async def http_download(
     server_info: ServerInfo,
     timeout: int,
     log: logging.Logger,
+    connect_timeout: int,
```

Comment:
What if we retained the single parameter but expanded the allowed type to include `aiohttp.ClientTimeout`?  Or perhaps required it.  Either way, it might improve flexibility and avoid a bit of passthrough?